redisson/src/main/java/org/redisson/cluster/ClusterConnectionManager.java

```diff
@@ -502,7 +502,7 @@ private ClusterPartition find(Collection<ClusterPartition> partitions, Integer s
         for (final ClusterPartition newPart : newPartitions) {
             boolean masterFound = false;
             for (ClusterPartition currentPart : getLastPartitions()) {
-                if (!newPart.getMasterAddress().equals(currentPart.getMasterAddress())) {
+                if (null != newPart && null != newPart.getMasterAddress() && !newPart.getMasterAddress().equals(currentPart.getMasterAddress())) {
```

Comment:
You're right about the under the the carpet thing. I' had open #761 with more element because i need help on this.

And i love yoda :)